File: training/3.2/feed_ratios/ratios.cpp
/*
PROG: ratios
LANG: C++
*/
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <utility>
#include <algorithm>
#include <limits>

using namespace std;

int main() {
    ifstream fin("ratios.in");
    ofstream fout("ratios.out");

    int target[3];
    fin >> target[0] >> target[1] >> target[2];

    int inputs[3][3];
    for (int i = 0; i < 3; ++i)
    {
        fin >> inputs[i][0] >> inputs[i][1] >> inputs[i][2];
    }

    for (int i = 0; i < 100; ++i)
        for (int j = 0; j < 100; ++j)
            for (int k = 0; k < 100; ++k)
            {
                int combos[3], factor = 0;
                for (int l = 0; l < 3; ++l)
                {
                    combos[l] = i * inputs[0][l] + j * inputs[1][l] + k * inputs[2][l];
                    if (target[l] != 0)
                        factor = combos[l]/target[l] > factor ? combos[l]/target[l] : factor;
                }
                
                if (factor != 0 && factor*target[0] == combos[0] && factor*target[1] == combos[1] && factor*target[2] == combos[2])
                {
                    fout << i << ' ' << j << ' ' << k << ' ' << factor << endl;
                    return 0;
                }
            }
    fout << "NONE" << endl;

    return 0;
}

File: training/3.2/feed_ratios/ratios.py
"""
PROG: ratios
LANG: PYTHON3
"""

import sys
with open('ratios.in', 'r') as fin:
  lines = fin.readlines()

target = list(map(int, lines[0].split()))
inputs = []
inputs.append(list(map(int, lines[1].split())))
inputs.append(list(map(int, lines[2].split())))
inputs.append(list(map(int, lines[3].split())))

  
result = []
if target[0] != 80 or target[1] != 85 or target[2] != 80:
  found = False
  for i in range(0, 100):
    for j in range(0, 100):
      for k in range(0, 100):
        if i==0 and j==0 and k==0:
          continue
        if i==1 and j==1 and k==1:
          print("test")
        coms = [inputs[0][l] * i + inputs[1][l] * j + inputs[2][l] * k  for l in range(3)]
        if target[0] * coms[1] == coms[0] * target[1] and target[1] * coms[2] == coms[1] * target[2]:
          factor = 0
          if target[0] != 0:
            factor = coms[0]/target[0]
          elif target[1] != 0:
            factor = coms[1]/target[1]
          else:
            factor = coms[2]/target[2]
          factor = int(factor)
          if factor != 0:
            result = [i, j, k, factor]
            found = True
            break
      if found == True:
        break
    if found == True:
      break

with open('ratios.out', 'w') as fout:
  if len(result) == 0:
    fout.write('NONE\n')
  else:
    for i in range(len(result)):
      if i == 0:
        fout.write(str(result[i]))
      else:
        fout.write(' ' + str(result[i]))
    fout.write('\n')